### ISSUE TITLE
typesenseSearchParameters.filter_by config doesn't apply in search results page

### ISSUE THREAD

**Issue by bbkgh** [typesense/docusaurus-theme-search-typesense#2]
Hi. When i set `filter_by:""` in typesenseSearchParameters, searching in modal works correctly but in search results page (show all X results) this option doesn't applies and accordingly returns nothing.

correct request in search modal:
```bash
curl 'https://typesense.mysite.com/multi_search?x-typesense-api-key=KEY' -X POST  -H 'Accept: application/json, text/plain, */*' -H 'Accept-Language: en-US,en;q=0.5' --compressed -H 'Content-Type: text/plain' -H 'Origin: http://localhost:3000' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'Referer: http://localhost:3000/' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: cross-site' -H 'TE: trailers' --data-raw '{"searches":[{"collection":"coll1","q":"kube","query_by":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","include_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content,anchor,url,type,id","highlight_full_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","group_by":"url","group_limit":3,"filter_by":""}]}'

```

incorrect request in search page (mysite.com/search?q=kube)
```bash
curl 'https://typesense.misite.com/multi_search?x-typesense-api-key=KEY' -X POST  -H 'Accept: application/json, text/plain, */*' -H 'Accept-Language: en-US,en;q=0.5' --compressed -H 'Content-Type: text/plain' -H 'Origin: http://localhost:3000' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'Referer: http://localhost:3000/' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: cross-site' -H 'TE: trailers' --data-raw '{"searches":[{"query_by":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","include_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content,anchor,url,type,id","highlight_full_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","group_by":"url","group_limit":3,"filter_by":"language:=[`en`] && docusaurus_tag:=[`default`,`docs-default-current`]","sort_by":"","collection":"coll","q":"kube","facet_by":"language,docusaurus_tag","page":1,"per_page":15},{"query_by":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","include_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content,anchor,url,type,id","highlight_full_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","group_by":"url","group_limit":3,"filter_by":"docusaurus_tag:=[`default`,`docs-default-current`]","sort_by":"","collection":"coll","q":"kube","facet_by":"language","page":1,"per_page":1},{"query_by":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","include_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content,anchor,url,type,id","highlight_full_fields":"hierarchy.lvl0,hierarchy.lvl1,hierarchy.lvl2,hierarchy.lvl3,hierarchy.lvl4,hierarchy.lvl5,hierarchy.lvl6,content","group_by":"url","group_limit":3,"filter_by":"language:=[`en`]","sort_by":"","collection":"coll","q":"kube","facet_by":"docusaurus_tag","page":1,"per_page":1}]}'
```

## Metadata

**Typsense Version**: 0.21.0

**Docusaurus Version**: 2.0.0-beta.0

**Plugin Version**: 0.0.5

**Comment by jasonbosco** [comment]
@bbkgh Thank you for bringing this up. I've pushed out a fix in `v0.1.0`. Could you confirm that it works fine now?

**Comment by damageboy** [comment]
Same happens here, I'll try and provide my own context:

The pages contain `docusaurus_tag` when indexing:
```html
....
<meta name="docusaurus_tag" content="docs-wiki-current" data-react-helmet="true">
...
```

The docsearch.json looks like this:

```json
{
  "index_name": "nextsilicon",
  "allowed_domains": [
     "nextsilicon-ui-public-docs"
   ],
  "start_urls": [
    "http://nextsilicon-ui-public-docs/"
  ],
  "sitemap_urls": [
    "http://nextsilicon-ui-public-docs/sitemap.xml"
  ],
  "sitemap_alternate_links": true,
  "stop_urls": [],
  "selectors": {
    "lvl0": {
      "selector": "(//ul[contains(@class,'menu__list')]//a[contains(@class, 'menu__link menu__link--sublist menu__link--active')]/text() | //nav[contains(@class, 'navbar')]//a[contains(@class, 'navbar__link--active')]/text())[last()]",
      "type": "xpath",
      "global": true,
      "default_value": "Documentation"
    },
    "lvl1": "header h1",
    "lvl2": "article h2",
    "lvl3": "article h3",
    "lvl4": "article h4",
    "lvl5": "article h5, article td:first-child",
    "lvl6": "article h6",
    "text": "article p, article li, article td:last-child"
  },
  "strip_chars": " .,;:#",
  "custom_settings": {
    "separatorsToIndex": "_",
    "attributesForFaceting": [
      "language",
      "version",
      "type",
      "docusaurus_tag"
    ],
    "attributesToRetrieve": [
      "hierarchy",
      "content",
      "anchor",
      "url",
      "url_without_anchor",
      "type"
    ]
  },
  "conversation_id": [
    "833762294"
  ],
  "nb_hits": 16653
}
```

When issuing a typesense collection query, I get the closest thing to a smoking gun with the list of fields:

```json
  {
    "created_at": 1630404189,
    "default_sorting_field": "item_priority",
    "fields": [
      {
        "facet": false,
        "index": true,
        "name": "anchor",
        "optional": true,
        "type": "string"
      },
      {
        "facet": false,
        "index": true,
        "name": "content",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "url",
        "optional": false,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "version",
        "optional": true,
        "type": "string[]"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl0",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl1",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl2",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl3",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl4",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl5",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "hierarchy.lvl6",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": ".*_tag",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "language",
        "optional": true,
        "type": "string"
      },
      {
        "facet": true,
        "index": true,
        "name": "tags",
        "optional": true,
        "type": "string[]"
      },
      {
        "facet": false,
        "index": true,
        "name": "item_priority",
        "optional": false,
        "type": "int64"
      }
    ],
    "name": "nextsilicon_1630403631",
    "num_documents": 0,
    "num_memory_shards": 4
  }
```

If you note, there is a super weird field there named... `"name": ".*_tag",` which seems to be a regex that would in theory include the `docusaurus_tag` but seems very weird that this would be the field name inside the typesense server?

**Comment by damageboy** [comment]
From what I see this is actually a typesense scraper issue of sorts...

The following code exists in `typesense-docsearch-scraper`:
https://github.com/typesense/typesense-docsearch-scraper/blob/7a2f05831bc5569e6c19e311125cb0ce91c61b1f/scraper/src/typesense_helper.py#L47:
```python
                {'name': '.*_tag', 'type': 'string', 'facet': True, 'optional': True},
```

This means that the actual field name is called `.*_tag`? Would this be enough for a query with `docusaurus_tag` to find the correct field?

**Comment by jasonbosco** [comment]
@damageboy `.*_tag` is a field with a regex name. I added it to to the schema to add support for tags like `docusaurus_tag` and for any other custom tag names that have `_tag`. Once you run the scraper once, it will then auto-create a field in the schema based on the actual name of the tag that exists in the markup. 

Do you see any JS errors in the browser console when the search page is blank?

**Comment by damageboy** [comment]
Well, I performed that query with the actual typesense scraped instance ...

Will test again

**Comment by damageboy** [comment]
The actual documents do not contain the `docusaurus_tag` field indexed:

```json
{
  "hierarchy": {
    "lvl0": "xxx",
    "lvl1": "xxx",
    "lvl2": null,
    "lvl3": null,
    "lvl4": null,
    "lvl5": null,
    "lvl6": null
  },
  "hierarchy.lvl0": "xxx",
  "hierarchy.lvl1": "xxx",
  "hierarchy_camel": [
    {
      "lvl0": "xxx",
      "lvl1": "xxx",
      "lvl2": null,
      "lvl3": null,
      "lvl4": null,
      "lvl5": null,
      "lvl6": null
    }
  ],
  "hierarchy_radio": {
    "lvl0": null,
    "lvl1": "xxx",
    "lvl2": null,
    "lvl3": null,
    "lvl4": null,
    "lvl5": null,
    "lvl6": null
  },
  "hierarchy_radio_camel": {
    "lvl0": null,
    "lvl1": "xxx",
    "lvl2": null,
    "lvl3": null,
    "lvl4": null,
    "lvl5": null,
    "lvl6": null
  },
  "id": "1",
  "item_priority": 90000,
  "no_variables": true,
  "objectID": "e18efb35b826a5bb8f163e0f29fd8b4e2c23915c",
  "tags": [],
  "type": "lvl1",
  "url": "http://localhost:8080/xxx/",
  "url_without_anchor": "http://localhost:8080/xxx/",
  "url_without_variables": "http://localhost:8080/xxx/",
  "weight": {
    "level": 90,
    "page_rank": 0,
    "position": 0
  }
}
```

While the html itself does contain it:

```html
...
<meta name="docusaurus_tag" content="docs-xxx-current" data-react-helmet="true">
```

**Comment by jasonbosco** [comment]
Hmmm, this used to be the case in an older version of the docsearch-scraper. Could you do a `docker pull typesense/docsearch-scraper` and then run the scraper again?

**Comment by octogonz** [comment]
Is this the same problem fixed by https://github.com/typesense/docusaurus-theme-search-typesense/pull/7 ?

**Comment by jasonbosco** [comment]
@octogonz Thank you for catching that!

@damageboy @bbkgh I've published @octogonz PR in `v0.3.0-0`. Could you give it a shot and let me know if that fixes the issue for you as well?

**Comment by octogonz** [comment]
I can confirm that my issue is solved by upgrading to `v0.3.0-0` (and redoing the crawl using the typesense-docsearch-scraper  tool). 👍👍  Thanks for the quick turnaround!

**Comment by jasonbosco** [comment]
Published `v0.3.0` with a few more fixes for Docusaurus 2.0.0-beta.15.
